tests/cases/test_raw_node.rs

```diff
@@ -269,6 +269,42 @@ fn test_raw_node_propose_add_duplicate_node() {
     assert_eq!(entries[2].take_data(), ccdata2);
 }
 
+#[test]
+fn test_raw_node_propose_add_learner_node() {
+    let s = new_storage();
+    let mut raw_node = new_raw_node(1, vec![], 10, 1, s.clone(), vec![new_peer(1)]);
+    let rd = raw_node.ready();
+    s.wl().append(&rd.entries).expect("");
+    raw_node.advance(rd);
+
+    raw_node.campaign().expect("");
+    loop {
+        let rd = raw_node.ready();
+        s.wl().append(&rd.entries).expect("");
+        if rd.ss.is_some() && rd.ss.as_ref().unwrap().leader_id == raw_node.raft.id {
+            raw_node.advance(rd);
+            break;
+        }
+        raw_node.advance(rd);
+    }
+
+    // propose add learner node and check apply state
+    let cc = conf_change(ConfChangeType::AddLearnerNode, 2);
+    raw_node.propose_conf_change(cc).expect("");
+
+    let rd = raw_node.ready();
+    s.wl().append(&rd.entries).expect("");
+    for e in rd.committed_entries.as_ref().unwrap() {
+        if e.get_entry_type() == EntryType::EntryConfChange {
```

Comment:
What if conf change entry is not found after iterating all entries?

Reply:
updated: check the committed_entries explicitly